pkg/deploy/controller/deployment/factory_test.go

```diff
@@ -0,0 +1,137 @@
+package deployment
```

Comment:
I think this can be greatly simplified. Remove this file and instead make a new test method in `controller_test.go` to exercise `makeDeployerPod` in isolation. The test could use test tables to assert that given a specific replicationController, an example pod is produced.